### DISCUSSION TITLE
Use regex to substitute text

### DISCUSSION THREAD

**Discussion by Grrishka** [Zettlr/Zettlr#2717] [Ideas]
Hi,

I enjoy regex.

If this awesome software could incorporate a classic regex system for search / substitution, that would be really great!

For example, it could allow me to fix [my title hierarchy problem](https://github.com/Zettlr/Zettlr/discussions/2716) without using another software!

     ;-)

grr.

**Comment by nathanlesage** [comment]
You can, in fact, use regular expressions to search and replace: https://docs.zettlr.com/en/core/search/#searching-in-files

**Reply by Grrishka** [reply]
Oh ! yes.

Thanks, I don't know why I didn't see that in the the doc ! I had tried a simple syntax (without the slashes of the JS syntax) and my quick search in the docs had only led me to the Regex for the IDs. 

I found a precise and not very talkative doc of this syntax on the [w3schools](https://www.w3schools.com/jsref/jsref_obj_regexp.asp).com website.

However, Zettlr takes a long time to perform searches and replacements on a whole document (not very long) and it doesn't replace special characters (notably \n, which I use to clean my markdown documents before exporting).

Incidentally, in the meantime, I've upgraded to version 2, which clearly implements Regex. So, obsolete idea, but the tool can probably be improved.

;)
